applications/connectivity_bridge/src/disk/config.c

```diff
@@ -0,0 +1,259 @@
+ */
+
+#include <zephyr.h>
+#include <fs/fs.h>
+#include <settings/settings.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define MODULE file_config
+#include "module_state_event.h"
+#include "fs_event.h"
+#include "ble_ctrl_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, LOG_LEVEL_DBG);
+
+#if CONFIG_FS_FATFS_LFN
+#define FILE_NAME         "Config.txt"
+#else
+#define FILE_NAME         "CONFIG.TXT"
+#endif
+
+#define SETTINGS_KEY "bridgecfg"
+
+#define SETTINGS_KEY_BLE "ble"
+#define SETTINGS_NAME_BLE "BLE_ENABLED"
+
+static const char file_contents_header[] =
+"==========================================\r\n"
+"  Nordic Thingy:91 Configuration options\r\n"
+"==========================================\r\n"
+"The parameters below can be changed at runtime.\r\n"
+"\r\n"
+"NOTE: For changes to take effect,\r\n"
+"safely disconnect (unmount) the drive and disconnect the USB cable.\r\n"
+"==========================================\r\n";
+
+static bool ble_enabled;
+
+static void send_ble_ctrl_event(bool enable)
+{
+	struct ble_ctrl_event *event = new_ble_ctrl_event();
+
+	event->cmd = enable ? BLE_CTRL_ENABLE : BLE_CTRL_DISABLE;
+	EVENT_SUBMIT(event);
+}
+
+static void save_settings(void)
+{
+	int err;
+
+	err = settings_save_one(
+			SETTINGS_NAME_BLE, &ble_enabled, sizeof(ble_enabled));
+	if (err) {
+		LOG_ERR("settings_save_one: %d", err);
+	}
+}
+
+static int settings_set(const char *key, size_t len_rd,
+			settings_read_cb read_cb, void *cb_arg)
+{
+	if (strcmp(key, SETTINGS_KEY_BLE) == 0) {
+		u32_t val;
+		ssize_t len = read_cb(cb_arg, &val, sizeof(val));
+
+		if (len == sizeof(val)) {
+			ble_enabled = val;
+		}
+	}
+
+	return 0;
+}
+
+static int commit(void)
+{
+	if (ble_enabled) {
+		send_ble_ctrl_event(true);
+	}
+	return 0;
+}
+
+static void settings_init(void)
+{
+	int err;
+
+	err = settings_subsys_init();
+	if (err) {
+		return;
+	}
+
+	static struct settings_handler sh = {
+		.name = SETTINGS_KEY,
+		.h_set = settings_set,
+		.h_commit = commit,
+	};
+
+	err = settings_register(&sh);
+	if (err) {
+		return;
+	}
+
+	err = settings_load();
+	if (err) {
+		return;
+	}
+}
+
+static bool read_line(struct fs_file_t *file, char *buf, size_t buf_size)
+{
+	ssize_t bytes_read;
+
+	bytes_read = fs_read(file, buf, buf_size - 1);
+	if (bytes_read <= 0) {
+		return false;
+	}
+
+	for (int i = 1; i < bytes_read; ++i) {
+		if (buf[i] == '\n') {
+			int err;
+
+			err = fs_seek(file, i - bytes_read + 1, FS_SEEK_CUR);
+			if (err) {
+				LOG_ERR("fs_seek: %d", err);
+				return false;
+			}
+
+			if (buf[i-1] == '\r') {
+				buf[i-1] = '\0';
+			} else {
+				buf[i] = '\0';
+			}
+
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static bool read_line_setting(char *buf, const char *key, char **setting)
+{
+	if (strncmp(buf, key, strlen(key)) != 0) {
+		return false;
+	}
+
+	for (int i = 0; i < (strlen(buf) - 1); ++i) {
+		if (buf[i] == '=') {
+			*setting = &buf[i + 1];
+			return true;
+		}
+	}
```

Comment:
char * eq_ptr = strchr (buf, '=');
if(eq_ptr) {
  *setting = eq_ptr + 1;
   return true;
}
 return false;

Reply:
Would in this case have to add some pointer arithmetic to ensure that *setting doesn't exceed string length in the event that '=' is the last character in the buffer

Reply:
Bah, okay, nevermind then.